artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/ActiveMQServerControlImpl.java

```diff
@@ -834,7 +835,12 @@ public int  getAddressMemoryUsagePercentage() {
 
    @Override
    public String getHAPolicy() {
-      return configuration.getHAPolicyConfiguration().getType().getName();
+      HAPolicyConfiguration haConfig = configuration.getHAPolicyConfiguration();
```

Comment:
should a default not be in the configuration rather than in the server control.
I think primary only is a sensible default, but I would also expect an embedded broker to provide configuration values that it cares about rather than depend on defaults. Defaults will only cover some subset at some time.

Reply:
It should avoid the NPE, and maybe return null if the policy is null.

Reply:
+1, returning null would reflect the truth (without depending on default behavior)